astropy/cosmology/funcs.py

```diff
@@ -113,33 +128,49 @@ def z_at_value(func, fval, zmin=1e-8, zmax=1000, ztol=1e-8, maxfun=500):
     other commonly inverted quantities) are monotonic in flat and open
     universes, but not in closed universes.
     """
-    from scipy.optimize import fminbound
+    from scipy.optimize import minimize_scalar
+
+    opt = {'maxiter': maxfun}
+    if method.lower() == 'bounded':
```

Comment:
Just something I noticed: If you allow method to be any callable won't this fail?

Reply:
Yes. I have not looked into custom methods and how exactly to use them, but I agree if they are mentioned in the docstring, they should not be made to immediately fail at least.